packages/admin/src/components/ImgUploadSlides/index.tsx

```diff
@@ -0,0 +1,217 @@
+import { imgType } from "src/types";
+
+import { LoadingSpinner, ToastMsg } from "../Atom";
+import ImgCard from "./ImgCard";
+import $ from "./style.module.scss";
+
+interface Props {
+  imgList: imgType[];
+}
+
+export default function ImgSlides({ imgList }: Props) {
+  const dispatch = useAppDispatch();
+  const { currentImgIdx } = useAppSelector(
+    (state) => state.boardReducer.board.write,
+  );
+  const idRef = useRef(-1);
+  const [ isFetched, setIsFetched ] = useState(false);
+  const [ imgSrcList, setImgSrcList ] = useState<imgType[]>([]);
+  const [ mouseDownClientX, setMouseDownClientX ] = useState<number>(0);
+  const [ mouseUpClientX, setMouseUpClientX ] = useState<number>(0);
+  const [ imgUpload, { isLoading, isSuccess } ] = useImgUploadMutation();
+  const loadingSpinner = (
+    <LoadingSpinner width={5} borderWidth={0.3} color="#000" />
+  );
+
+  useEffect(() => {
+    const dragSpace = Math.abs(mouseDownClientX - mouseUpClientX);
+
+    if (mouseDownClientX !== 0) {
+      if (mouseUpClientX < mouseDownClientX && dragSpace > 100) {
+        dispatch(changeCurrentImg(currentImgIdx + 1));
+      } else if (mouseUpClientX > mouseDownClientX && dragSpace > 100) {
+        dispatch(changeCurrentImg(currentImgIdx - 1));
+      }
+    }
+  }, [ mouseUpClientX ]);
+
+  useEffect(() => {
+    if (isFetched) {
+      setTimeout(() => setIsFetched(false), 1000);
+    }
+  }, [ isFetched ]);
+
+  const uploadFiles = async (files: FileList) => {
+    const formData = new FormData();
+    if (files) {
+      const filesArr = Array.from(files);
+      filesArr.forEach((file: File) => {
+        formData.append("image", file);
+      });
+
+      const res = [ ...imgListMocks ]; // 서버 통신이 안돼서 mock데이터로 테스트
+      const data = res.map((src) => ({
+        id: idRef.current++,
+        src,
+      }));
+      setIsFetched(true);
+      setImgSrcList([ ...imgSrcList, ...data ]);
+      dispatch(writeBoard({ boardImgList: [ ...imgSrcList, ...data ] }));
+
+      // try {
+      //   const res = await imgUpload(formData).unwrap();
+      //   const data = res.map((src) => {
+      //     idRef.current += 1;
+      //     return {
+      //       id: idRef.current + 1,
+      //       src,
+      //     };
+      //   });
+      //   setIsFetched(true);
+      //   setImgSrcList([ ...imgSrcList, ...data ]);
+      //   dispatch(writeBoard({ boardImgList: [ ...imgSrcList, ...data ] }));
+      // } catch (e) {
+      //   console.log(e);
+      // } finally {
+      //   setIsFetched(true);
+      // }
+    }
+  };
+
+  function onLoadFile(e: React.ChangeEvent | React.DragEvent) {
+    if (e.type === "drop") {
+      const { dataTransfer } = e as React.DragEvent<HTMLInputElement>;
+      uploadFiles(dataTransfer.files);
+    }
+    if (e.type === "change") {
+      const { target } = e as React.ChangeEvent<HTMLInputElement>;
+      if (target.files) uploadFiles(target.files);
+    }
+  }
```

Comment:
말씀해주신 타입 단언 안 쓰는 방법을 생각해봤는데 혹시 이 방법은 어떨지 여쭤봅니다...!
```suggestion
  function onLoadFile(e: React.ChangeEvent | React.DragEvent) {
    if (e.type === "drop" && "dataTransfer" in e) {
      const {
        dataTransfer: { files },
      } = e;
      uploadFiles(files);
    }
    if (e.type === "change" && "files" in e.target) {
      const {
        target: { files },
      } = e;
      if (files) uploadFiles(files);
    }
  }
```

Reply:
102번 줄에 return을 넣어주신 이유가 있을까요??

Reply:
함수 안 조건문에서 return 쓰는 게 습관이 되었나 봅니다 😂